### ISSUE TITLE
Bug Report EvalWrapper

### ISSUE THREAD

**Issue by Vedaevolution** [Microsoft/CNTK#1186]
Dear Developers,

i know it has already been written somewhere in another issue but i want to confirm it, because i don't think it's a mistake made by programmers using the wrapper. I'm using the RCNN example and was observing a very strange behaviour. I trained a model via GPU which worked fine and had a good mAP, but when i tried to use the trained network on the very same samples it was trained on with the wrapper, it failed 100% of the time. The class it would have been predicted would always have been background which clearly was not the case.

I'm using the CNTK 2.0 beta 5.  At first i thought i did something wrong and double checked everything but i could not find a mistake. 

Finally i tried using the CNTK 2.0 beta 3 wrapper on my already with CNTK 2.0 beta 5 trained model and it worked without any problem. Because i did not change anything in my project and just switched beta 5 with beta 3 i really think this is probably a bug. 

The exact same code which gives garbage under beta 5 works under beta 3 like a charm. Is it possible that there is a calculation error using the CPU and can you confirm it? If you wish i can create a C# project reproducing the error, at least on my machine and send it to you. 

Best Regards
Stephen

**Comment by zhouwangzw** [comment]
@Stephen, is your model trained using Batch Normalization engine? If yes, is useCntkEngine set to true or false (if false, it uses CuDNN engine). The EvalWrapper only works on CPU, so it does not support BN using CuDNN engine. 
Could you please provide the configuration (BrainScript or Python?) you used to train the model? We will take a look into it.

**Comment by Vedaevolution** [comment]
@zhouwangzw, as far as i know i don't use it.  Here is the config.


[fastrcnn.zip](https://github.com/Microsoft/CNTK/files/666557/fastrcnn.zip)

**Comment by zhouwangzw** [comment]
You are using the pre-trained model:
`network     = BS.Network.Load ("../../../../../PretrainedModels/AlexNet.model")
` 
How was this model trained?
If you could create a project for repo, we can take a look into it. 
Btw., we are going to publish a new Beta version this week, and you might want to check whether the issue exists in the new version. 
Thanks.

**Comment by Vedaevolution** [comment]
Well i was basically using this example: [https://github.com/Microsoft/CNTK/wiki/Object-Detection-using-Fast-R-CNN](url)
The problem should be reproducable by simply using the Microsoft standard example i linked here and let it train on some samples. After that the results using EvalWrapper should be incorrect on the very train data itself.

Concerning the AlexModel i don't think it uses BN but I am not sure. It's a model delivered from this site. You can get it using the linke above. If i'm wrong please correct me. 

Thank you very much for your effort. Even if the AlexModel uses BN it would be nice to have this info in the Tutorial for Fast RCNN because i can't find it there.

I will try the new version as soon as it comes out. :)

Best Regards
Stephen

**Comment by Vedaevolution** [comment]
Seems my issue has been found and will be solved: [https://github.com/Microsoft/CNTK/issues/1154](url)

:)

**Comment by zhouwangzw** [comment]
I see. It seems that there is a bug in the CPU convolution path. We will provide a fix soon.
I am closing this issue, but leave #1154 open until the problem is fixed. Please re-open it if needed.
